### DISCUSSION TITLE
Support plan for Hierarchy Tree Diagrams

### DISCUSSION THREAD

**Discussion by kishcs** [plantuml/plantuml#1169] [Ideas]
Good to support Hierarchy Tree Diagrams. 
Use-case: Family hierarchy

**Comment by fpolowood** [comment]
Here is a library for family hierarchy with the current PlantUML digraph support. 
Library documentation: https://gitlab.migrandama.com/foss/plantuml-templates
[Check out a sample](https://www.plantuml.com/plantuml/svg/dPBTQiCW7CNlynJE3TQ1hSJBGQ7ieeN0O1Tx0Hkr4Vm8HYbbxDrdT6dicglb8ia_7iyvl-1kyDJve1NOKotLvTDnNZq3aOQrZnDGykYekm1csw2vyPm1wa7dVJ-yOdoINj7ZHikJeuPHJJUjrLZOOS2zecQqNdkkuxldG_KWJQi2u__RZjVOqJFML1fSDDpqKLMnqQBlm05yLG14bzu6mxRGky1X-X9wnLSffwSkWWmrWeTYX0W-YdWn0T6lh4PxwZlkygpZ_6x7-MNPa2mPCa5dG_B7a1I6Ze-KAv44DRffMgnhUxzEYa8ptCTSWmaMoI3AvbUix3LH5VpNuaLm5uPZZ4HJm7qcoPa5EhjcajjCCcK-fKVRISNYpYri0ZK8D459MuQVfU1eWJQV0mJpySse6M6gsrYEqkv6AZN_YgbWHapvgtIlqK4wXIRZg1SBAeBUx6L1z5tjk651gny0)

**Reply by VladimirAlexiev** [reply]
Hi @fpolowood and thanks for your contribution!
However, both the example above and https://gitlab.migrandama.com/foss/plantuml-templates/-/blob/main/org-chart.md show error "Cannot open URL".
No clue what is the problem since https://gitlab.migrandama.com/foss/plantuml-templates/-/raw/main/family-tree.puml opens fine...

**Comment by The-Lum** [comment]
See also this other _"PlantUML extension for genealogy trees automatic generation" of @zhmylove_:
- https://github.com/zhmylove/genealogy

Regards.

**Reply by rafgugi** [reply]
just saw this, but when I check there is an issue https://github.com/zhmylove/genealogy/issues/3

now Im not sure what the expected result of zhmylove/genealogy 🤔

**Comment by fpolowood** [comment]
Hi Vladimir, 

It took me a long time to troubleshoot. The code is just fine, but unfortunately, the change in behavior of plantuml server introduced here ( [ https://plantuml.com/security | https://plantuml.com/security ] ) means that you can no longer include files that are not specifically in the 'allowlist', which makes include useless except for self hosted servers. It is a shame, in the name of security this was a regression that discourages reuse, forcing people to include all the code on every single diagram. 

Felipe

**Reply by zhmylove** [reply]
@fpolowood Nice catch!
Is it possible to store the code on github pages and include raw version via url?

**Comment by VladimirAlexiev** [comment]
https://github.com/plantuml/plantuml-server/issues/327

**Comment by kishcs** [comment]
Can this directly be processed by Plantuml ? 
 https://gitlab.migrandama.com/foss/plantuml-templates/-/raw/main/family-tree.puml
